homeassistant/core.py

```diff
@@ -809,21 +810,38 @@ def _gen_unique_id():
 
     @property
     def services(self):
-        """Return dictionary with per domain a list of available services."""
+        """Return dictionary with per domain a dict of available services."""
         return run_callback_threadsafe(
             self._hass.loop, self.async_services,
         ).result()
 
     @callback
     def async_services(self):
-        """Return dictionary with per domain a list of available services.
+        """Return dictionary with per domain a dict of available services.
 
         This method must be run in the event loop.
         """
         return {domain: {key: value.as_dict() for key, value
                          in self._services[domain].items()}
                 for domain in self._services}
 
+    @property
+    def full_services(self):
```

Comment:
I don't think that we should introduce new sync APIs unless there is code using it. I also don't perse see a reason to have two properties doing pretty much the same. Why not update the old one to just return the service objects?

Reply:
I think it's currently used at least by api.py that requires json-serializable only. I guess we could move that conversion to api.py.